lib/plugins/cleaner.rb

```diff
@@ -0,0 +1,51 @@
+require 'rubygems/package'
+require_relative '../moonshot/creds_helper'
+
+module Moonshot
+  module Plugins
+    # Moonshot plugin class for removing artifacts when a stage gets deleted.
+    class Cleaner
+      include Moonshot::CredsHelper
+
+      def initialize(bucket_name)
+        @bucket_name = bucket_name
+      end
+
+      def post_delete(resources)
+        stack_name = resources.stack.name
+        resources.ilog.start 'Starting to delete artifacts for stack: ' \
+          "#{stack_name}" do |s|
+          arts = retrieve_artifact_names(stack_name)
+          if arts.empty?
+            s.success 'No artifacts found, nothing deleted.'
+          else
+            delete_artifacts arts
+            s.success "Deleted artifacts for stack: '#{stack_name}' " \
+                      "in bucket: '#{@bucket_name}'"
+          end
+        end
+      end
+
+      private
+
+      def retrieve_artifact_names(name)
+        objs = s3_client.list_objects(bucket: @bucket_name, prefix: name)
+        valid_keys = []
+        objs.contents.each do |o|
+          valid_keys << o.key if o.key =~ /^#{name}[\.-]/
```

Comment:
Builds aren't actually related to a stack though, so putting them into a
directory doesn't make sense for moving through environments in a CI
workload. You would create a named build in dev, move through staging and
load testing environments for example then deploy to prod. That all has
nothing to do with the original stack it was built for.

On Thu, Jul 28, 2016 at 10:36 AM Chadderton Odwazny <
notifications@github.com> wrote:

> In lib/plugins/cleaner.rb
> <https://github.com/acquia/moonshot/pull/114#discussion_r72633912>:
>
> > +            s.success 'No artifacts found, nothing deleted.'
> > +          else
> > +            delete_artifacts arts
> > +            s.success "Deleted artifacts for stack: '#{stack_name}' " \
> > +                      "in bucket: '#{@bucket_name}'"
> > +          end
> > +        end
> > +      end
> > +
> > +      private
> > +
> > +      def retrieve_artifact_names(name)
> > +        objs = s3_client.list_objects(bucket: @bucket_name, prefix: name)
> > +        valid_keys = []
> > +        objs.contents.each do |o|
> > +          valid_keys << o.key if o.key =~ /^#{name}[\.-]/
>
> That does sound like a good idea. I wonder why we didn't use a directory
> to begin with? Perhaps @askreet <https://github.com/askreet> remembers?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/acquia/moonshot/pull/114/files/9f5207a59452b534f809307b66a7d5d254ff3750#r72633912>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAodKOGw55pg_RYaD7hZJ_U3yx307e0nks5qaL5IgaJpZM4JWRXT>
> .
>